tools/codesign_mac_executable.py

```diff
@@ -0,0 +1,232 @@
+    cmd_result = subprocess.run(cmd)
+    assert cmd_result.returncode == 0
+
+    cmd = [
+        "install_name_tool",
+        "-rpath",
+        "/Volumes/Work/Perforce/UE4/Engine/Binaries/ThirdParty/PhysX3/Mac",
+        os.path.join(executable_name, "Contents", "UE4", "Engine", "Binaries", "ThirdParty", "PhysX3", "Mac"),
+        os.path.join(executable_name, "Contents", "UE4", "Engine", "Binaries", "ThirdParty", "PhysX3", "Mac", "libAPEX_Legacy.dylib")]
+    print(f"[SPEAR | codesign_mac_executable.py] Executing: {' '.join(cmd)}")
+    cmd_result = subprocess.run(cmd)
+    assert cmd_result.returncode == 0
+
+    cmd = [
+        "install_name_tool",
+        "-rpath",
+        "/Volumes/Work/Perforce/UE4/Engine/Binaries/ThirdParty/PhysX3/Mac",
+        os.path.join(executable_name, "Contents", "UE4", "Engine", "Binaries", "ThirdParty", "PhysX3", "Mac"),
+        os.path.join(executable_name, "Contents", "UE4", "Engine", "Binaries", "ThirdParty", "PhysX3", "Mac", "libApexFramework.dylib")]
+    print(f"[SPEAR | codesign_mac_executable.py] Executing: {' '.join(cmd)}")
+    cmd_result = subprocess.run(cmd)
+    assert cmd_result.returncode == 0
+
+    # revert current working directory
+    os.chdir(cwd)
+
+    # files that need to be codesigned
+    files = [
+        os.path.join(args.macos_executable, "Contents", "UE4", "Engine", "Binaries", "ThirdParty", "Ogg", "Mac", "libogg.dylib"),
+        os.path.join(args.macos_executable, "Contents", "UE4", "Engine", "Binaries", "ThirdParty", "Vorbis", "Mac", "libvorbis.dylib"),
+        os.path.join(args.macos_executable, "Contents", "UE4", "Engine", "Binaries", "ThirdParty", "PhysX3", "Mac", "libPhysX3.dylib"),
+        os.path.join(args.macos_executable, "Contents", "UE4", "Engine", "Binaries", "ThirdParty", "PhysX3", "Mac", "libPhysX3Common.dylib"),
+        os.path.join(args.macos_executable, "Contents", "UE4", "Engine", "Binaries", "ThirdParty", "PhysX3", "Mac", "libPxFoundation.dylib"),
+        os.path.join(args.macos_executable, "Contents", "UE4", "Engine", "Binaries", "ThirdParty", "PhysX3", "Mac", "libAPEX_Clothing.dylib"),
+        os.path.join(args.macos_executable, "Contents", "UE4", "Engine", "Binaries", "ThirdParty", "PhysX3", "Mac", "libPxPvdSDK.dylib"),
+        os.path.join(args.macos_executable, "Contents", "UE4", "Engine", "Binaries", "ThirdParty", "PhysX3", "Mac", "libNvCloth.dylib"),
+        os.path.join(args.macos_executable, "Contents", "UE4", "Engine", "Binaries", "ThirdParty", "PhysX3", "Mac", "libPhysX3Cooking.dylib"),
+        os.path.join(args.macos_executable, "Contents", "UE4", "Engine", "Binaries", "ThirdParty", "PhysX3", "Mac", "libAPEX_Legacy.dylib"),
+        os.path.join(args.macos_executable, "Contents", "UE4", "Engine", "Binaries", "ThirdParty", "PhysX3", "Mac", "libApexFramework.dylib"),
+        os.path.join(args.macos_executable, "Contents", "MacOS", os.path.splitext(executable_name)[0])]
```

Comment:
we refer to Contents/MacOS differently in env.py, let's do the same thing here for consistency

Reply:
I like this better, and maybe we should change the way we are handling this in env.py, because in env.py we are unnecessarily complicating the code to get the path to the executable.
What do you think?

Reply:
Ok sure, then let's change env.py. Here, you're assuming that the "internal" executable name is the same as the app name, whereas env.py doesn't make that assumption. But the way you're doing it here is more straightforward than env.py.

Reply:
Updated env.py